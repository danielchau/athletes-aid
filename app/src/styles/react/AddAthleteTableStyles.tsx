import { makeStyles, Theme, createStyles } from "@material-ui/core/styles";

/**
 * React styles for the Add Athletes Table.
 */
export const addAthleteTableStyles = makeStyles((theme: Theme) =>
    createStyles({
        root: {
            width: "100%",
            padding: "8px",
            height: "100%",
            display: "flex",
            justifyContent: "center"
        },
        tableContainer: {
            width: "100%",
            height: "100%",
            padding: "8px",
            backgroundColor: "rgba(0, 33, 69, 0.05)",
            overflow: "scroll"
        },
        table: {
            width: "100%"
        },
        tableBodyContainer: {
            width: "2000px",
            overflow: "auto",
            height: "100%"
        },
        tableBody: {
            tableLayout: "fixed"
        },
        tableRow: {
            cursor: "pointer"
        },
        tooltip: {
            backgroundColor: "#fff",
            color: "#4f4f4f",
            boxShadow: theme.shadows[1]
        }
    })
);
