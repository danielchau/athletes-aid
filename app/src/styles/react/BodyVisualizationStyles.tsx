import { makeStyles, Theme, createStyles } from "@material-ui/core/styles";

/**
 * React styles for the Body Visualization.
 */
export const bodyVisualizationStyles = makeStyles((theme: Theme) =>
    createStyles({
        root: {
            width: "100%",
            height: "100%"
        },
        imageContainer: {
            position: "relative",
            alignSelf: "center"
        },
        image: {
            width: "100%"
        },
        dot: {
            height: "15px",
            width: "15px",
            position: "absolute",
            borderRadius: "15px"
        },
        legendContainer: {
            borderLeft: "#dedede solid 1px"
        },
        legendItem: {
            display: "flex",
            color: "rgba(0, 0, 0, 0.54)",
            marginBottom: "2px"
        },
        dotLegend: {
            height: "10px",
            width: "10px",
            borderRadius: "10px",
            marginTop: "6px",
            marginRight: "4px"
        }
    })
);
