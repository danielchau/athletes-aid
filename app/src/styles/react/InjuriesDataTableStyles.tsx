import { makeStyles, Theme, createStyles, lighten } from "@material-ui/core/styles";

/**
 * React styles for the Injury Data Table.
 */
export const injuriesDataTableStyles = makeStyles((theme: Theme) =>
    createStyles({
        root: {
            width: "100%",
            marginTop: theme.spacing(3)
        },
        paper: {
            width: "100%",
            marginBottom: theme.spacing(2),
            backgroundColor: "rgba(0, 33, 69, 0.05)"
        },
        table: {
            minWidth: 750
        },
        tableWrapper: {
            overflowX: "auto"
        },
        tableRow: {
            cursor: "pointer"
        },
        visuallyHidden: {
            border: 0,
            clip: "rect(0 0 0 0)",
            height: 1,
            margin: -1,
            overflow: "hidden",
            padding: 0,
            position: "absolute",
            top: 20,
            width: 1
        }
    })
);

/**
 * React styles for the Injury Data Table Toolbar Header.
 */
export const injuriesDataTableToolbarStyles = makeStyles((theme: Theme) =>
    createStyles({
        root: {
            paddingLeft: theme.spacing(2),
            paddingRight: theme.spacing(1)
        },
        highlight:
            theme.palette.type === "light"
                ? {
                      color: theme.palette.primary.main,
                      backgroundColor: lighten(theme.palette.primary.light, 0.85)
                  }
                : {
                      color: theme.palette.text.primary,
                      backgroundColor: theme.palette.secondary.dark
                  },
        title: {
            flex: "1 1 100%"
        },
        export: {
            margin: "auto",
            color: theme.palette.secondary.main,
            display: "flex"
        }
    })
);
