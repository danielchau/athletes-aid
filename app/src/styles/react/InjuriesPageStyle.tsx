import { makeStyles, Theme, createStyles } from "@material-ui/core/styles";

/**
 * React styles for the Injuries Page.
 */
export const injuriesPageStyles = makeStyles((theme: Theme) =>
    createStyles({
        root: {
            padding: "16px",
            backgroundColor: "#fff"
        },
        drawerOpen: {
            width: "calc(100vw - 240px)",
            transition: theme.transitions.create("width", {
                easing: theme.transitions.easing.sharp,
                duration: theme.transitions.duration.enteringScreen
            }),
            [theme.breakpoints.down("xs")]: {
                width: "100vw"
            }
        },
        drawerClosed: {
            width: "calc(100vw - 72px)",
            transition: theme.transitions.create("width", {
                easing: theme.transitions.easing.sharp,
                duration: theme.transitions.duration.leavingScreen
            }),
            [theme.breakpoints.down("xs")]: {
                width: "100vw"
            }
        },
        grid: {
            backgroundColor: "#fff"
        },
        paper: {
            padding: theme.spacing(2),
            textAlign: "center",
            color: theme.palette.text.secondary,
            backgroundColor: "rgba(0, 33, 69, 0.05)",
            height: "100%"
        },
        vizPaper: {
            padding: theme.spacing(2),
            height: "100%"
        },
        primaryStatisticContainer: {
            display: "flex",
            flexDirection: "column"
        },
        primaryStatisticValue: {
            overflow: "hidden",
            textOverflow: "ellipsis",
            fontSize: "calc((100vw - 240px)/16)"
        },
        primaryStatisticLabel: {
            paddingTop: "16px"
        },
        dateTimeContainer: {},
        textField: {
            marginLeft: theme.spacing(1),
            marginRight: theme.spacing(1),
            width: 200
        },
        dateField: {
            width: 200,
            marginRight: theme.spacing(1),
            marginTop: 0,
            marginBottom: 0
        },
        dateTimeButton: {
            marginLeft: "8px",
            marginTop: "6px"
        }
    })
);
