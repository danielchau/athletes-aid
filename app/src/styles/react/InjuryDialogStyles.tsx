import { makeStyles, Theme, createStyles } from "@material-ui/core/styles";

export const injuryDialogStyles = makeStyles((theme: Theme) =>
    createStyles({
        appBar: {
            position: "relative"
        },
        title: {
            marginLeft: theme.spacing(2),
            flex: 1
        },
        dialogTitle: {
            padding: "0"
        },
        dialogPaper: {
            height: "100%",
            maxWidth: "600px",
            width: "50%",
            minWidth: "200px",
            position: "fixed",
            bottom: "0",
            right: "0",
            margin: "0"
        },
        dialogContentContainer: {
            padding: "0"
        },
        dialogContent: {
            padding: "16px"
        },
        dialogContentPaper: {
            backgroundColor: "#fafafa",
            padding: "8px",
            marginBottom: "8px"
        },
        notePaper: {
            backgroundColor: "#ffffff",
            padding: "8px",
            marginBottom: "8px"
        },
        noteContainer: {
            display: "flex",
            flexDirection: "row"
        },
        noteIcon: {
            margin: "auto",
            marginLeft: "8px",
            marginRight: "16px"
        },
        newNoteContainer: {
            display: "flex",
            flexDirection: "row",
            width: "100%",
            justifyContent: "center",
            alignItems: "center",
            alignSelf: "center"
        },
        newNote: {
            flexGrow: 1,
            paddingRight: "8px"
        },
        sendButton: {
            color: "#F2A71E"
        },
        activeSwitchContainer: {
            display: "flex",
            flexDirection: "row",
            justifyContent: "center",
            alignContent: "center",
            alignSelf: "center",
            paddingBottom: "8px"
        }
    })
);
