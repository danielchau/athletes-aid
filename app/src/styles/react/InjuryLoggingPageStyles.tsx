import { makeStyles, Theme, createStyles } from "@material-ui/core/styles";

export const injuryLoggingPageStyles = makeStyles((theme: Theme) =>
    createStyles({
        root: {
            width: "100%",
            padding: "8px"
        },
        backButton: {
            marginRight: theme.spacing(1)
        },
        completedContainer: {
            height: "calc(100% - 116px)",
            textAlign: "center",
            paddingTop: "20%"
        },
        instructions: {
            marginTop: theme.spacing(1),
            marginBottom: theme.spacing(1)
        },
        paper: {
            color: theme.palette.text.secondary,
            backgroundColor: "#fafafa"
        },
        paperContent: {
            height: "100%",
            padding: "16px",
            paddingRight: "8px",
            overflow: "auto"
        },
        stepper: {
            backgroundColor: "#fafafa",
            borderRadius: "4px"
        },
        loggingContent: {
            height: "calc(100% - 168px)",
            marginBottom: "52px",
            marginTop: "8px",
            overflow: "hidden"
        },
        loggingBottomButtons: {
            position: "fixed",
            bottom: 0,
            right: 0,
            height: "52px",
            padding: "8px"
        },
        multilineInput: {
            width: "100%"
        }
    })
);
