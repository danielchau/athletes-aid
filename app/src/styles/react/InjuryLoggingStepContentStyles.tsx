import { makeStyles, Theme, createStyles } from "@material-ui/core/styles";

export const injuryLoggingStepContentStyles = makeStyles((theme: Theme) =>
    createStyles({
        formControl: {
            minWidth: 200,
            width: "50%",
            paddingRight: "8px",
            paddingBottom: "16px"
        },
        textInput: {
            minWidth: 200,
            width: "100%",
            paddingRight: "8px",
            paddingBottom: "16px"
        },
        multilineInput: {
            width: "100%",
            paddingBottom: "16px",
            margin: "0px"
        },
        dateInput: {
            minWidth: 200,
            width: "50%",
            margin: "0px",
            paddingRight: "8px",
            paddingBottom: "16px"
        },
        checkboxInput: {
            minWidth: 200,
            width: "50%",
            margin: "0px",
            paddingRight: "8px",
            paddingBottom: "23px",
            paddingTop: "7px"
        },
        instructions: {
            textAlign: "center",
            marginBottom: theme.spacing(3),
            color: theme.palette.text.secondary,
            width: "100%"
        }
    })
);
