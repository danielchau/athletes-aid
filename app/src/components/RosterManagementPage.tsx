import * as React from "react";
import { rosterManagementPageStyles } from "../styles/react/RosterManagementPageStyles";
import Paper from "@material-ui/core/Paper";
import FormControl from "@material-ui/core/FormControl";
import Select from "@material-ui/core/Select";
import InputLabel from "@material-ui/core/InputLabel";
import MenuItem from "@material-ui/core/MenuItem";
import Typography from "@material-ui/core/Typography";
import Button from "@material-ui/core/Button";
import IconButton from "@material-ui/core/IconButton";
import TextField from "@material-ui/core/TextField";
import List from "@material-ui/core/List";
import ListItem from "@material-ui/core/ListItem";
import ListItemSecondaryAction from "@material-ui/core/ListItemSecondaryAction";
import ListItemText from "@material-ui/core/ListItemText";
import ListItemAvatar from "@material-ui/core/ListItemAvatar";
import Checkbox from "@material-ui/core/Checkbox";
import Avatar from "@material-ui/core/Avatar";
import Divider from "@material-ui/core/Divider";
import DeleteIcon from "@material-ui/icons/Delete";
import AddIcon from "@material-ui/icons/Add";
import { Athlete, Team } from "../util/types";
import MyDropzone from "./Dropzone";

interface RosterManagementPageProps {
    teams: Team[];
}

export default function RosterManagementPage(props: RosterManagementPageProps) {
    const classes = rosterManagementPageStyles({});
    const [selectedTeam, setSelectedTeam] = React.useState<Team | null>(null);
    const [teamName, setTeamName] = React.useState<string | null>(null);
    const [season, setSeason] = React.useState<string | null>(null);
    const [checked, setChecked] = React.useState(new Set<string>());
    const [newAthletes, setNewAthletes] = React.useState<Athlete[]>([]);

    const handleTeamSelected = (
        event: React.ChangeEvent<{ value: string }>
    ) => {
        let team = props.teams.filter(team => team.id === event.target.value);
        setSelectedTeam(team.length > 0 ? team[0] : null);
        setTeamName(null);
    };

    const handleToggle = (value: string) => () => {
        const newChecked = new Set(checked);

        if (checked.has(value)) {
            newChecked.delete(value);
        } else {
            newChecked.add(value);
        }

        setChecked(newChecked);
    };

    const handleAddTeam = () => {
        setSelectedTeam(null);
        setTeamName(null);
    };

    const handleTeamNameChange = (
        event: React.ChangeEvent<{ value: string }>
    ) => {
        setTeamName(event.target.value);
    };

    const handleSeasonChange = (
        event: React.ChangeEvent<{ value: string }>
    ) => {
        setSeason(event.target.value);
    };

    const handleAthleteDelete = () => {};

    const handleAddAthletes = () => {};

    const handleSave = () => {};

    return (
        <div className={classes.root}>
            <Paper className={classes.introPaper}>
                <Typography className={classes.introText}>
                    Edit one of your current teams:
                </Typography>
                <FormControl className={classes.formControl} variant="outlined">
                    <InputLabel id="team-select-label">Team</InputLabel>
                    <Select
                        labelWidth={40}
                        id="team-select"
                        value={!!selectedTeam ? selectedTeam.id : ""}
                        onChange={handleTeamSelected}
                    >
                        <MenuItem value="" disabled>
                            Select the team to edit...
                        </MenuItem>
                        {props.teams.map((team: Team, i: number) => (
                            <MenuItem key={i} value={team.id}>
                                {team.name}
                            </MenuItem>
                        ))}
                    </Select>
                </FormControl>
                <Typography className={classes.introText}>
                    or add a new one:
                </Typography>
                <IconButton
                    aria-label="add"
                    className={classes.introButton}
                    onClick={handleAddTeam}
                >
                    <AddIcon />
                </IconButton>
            </Paper>
            <Paper className={classes.paperContent}>
                <div>
                    <TextField
                        id="team-name"
                        className={classes.textInput}
                        label="Team Name"
                        placeholder="(e.g. Women's Soccer)"
                        margin="normal"
                        variant="outlined"
                        value={
                            !!selectedTeam
                                ? teamName != null
                                    ? teamName
                                    : selectedTeam.name
                                : ""
                        }
                        onChange={handleTeamNameChange}
                    />
                    <TextField
                        id="season"
                        className={classes.textInput}
                        label="Season"
                        placeholder="(e.g. 2019/2020)"
                        margin="normal"
                        variant="outlined"
                        value={
                            !!selectedTeam
                                ? season != null
                                    ? season
                                    : selectedTeam.season
                                : ""
                        }
                        onChange={handleSeasonChange}
                    />
                </div>
                <Divider className={classes.contentDivider} />
                <div className={classes.athleteContentContainer}>
                    <Paper className={classes.card}>
                        <div className={classes.athletesContainer}>
                            <List dense className={classes.athletesList}>
                                {!!selectedTeam ? (
                                    selectedTeam.athletes.map(
                                        (athlete: Athlete) => {
                                            const labelId = `checkbox-list-secondary-label-${athlete.name}`;
                                            return (
                                                <ListItem
                                                    key={athlete.id}
                                                    button
                                                >
                                                    <ListItemAvatar>
                                                        <Avatar
                                                            alt={`Avatar nÂ°${athlete.id}`}
                                                        >
                                                            {athlete.name.slice(
                                                                0,
                                                                1
                                                            )}
                                                        </Avatar>
                                                    </ListItemAvatar>
                                                    <ListItemText
                                                        id={labelId}
                                                        primary={athlete.name}
                                                    />
                                                    <ListItemSecondaryAction>
                                                        <Checkbox
                                                            edge="end"
                                                            onChange={handleToggle(
                                                                athlete.id
                                                            )}
                                                            checked={checked.has(
                                                                athlete.id
                                                            )}
                                                            inputProps={{
                                                                "aria-labelledby": labelId
                                                            }}
                                                        />
                                                    </ListItemSecondaryAction>
                                                </ListItem>
                                            );
                                        }
                                    )
                                ) : (
                                    <></>
                                )}
                            </List>
                            <Divider light />
                            <Button
                                variant="contained"
                                className={classes.existingAthletesButton}
                                onClick={handleAthleteDelete}
                            >
                                Delete
                                {checked.size > 0
                                    ? " (" + checked.size + " selected)"
                                    : ""}
                                <DeleteIcon />
                            </Button>
                        </div>
                    </Paper>
                    <Paper className={classes.card}>
                        <div className={classes.athletesContainer}>
                            <div className={classes.uploadPrompt}>
                                <div className={classes.fileDownload}>
                                    <Typography>
                                        Please upload the filled spreadsheet
                                        found{" "}
                                        <a href="" download>
                                            here
                                        </a>{" "}
                                        and then add the athletes.
                                    </Typography>
                                </div>
                                <Divider light />
                                <div className={classes.dropzone}>
                                    <MyDropzone></MyDropzone>
                                </div>
                                <Divider light />
                                <div className={classes.addedAthletes}>
                                    Athletes
                                </div>
                            </div>
                            <div>
                                <Button
                                    variant="contained"
                                    disabled={newAthletes.length == 0}
                                    className={classes.newAthletesButton}
                                    onClick={handleAddAthletes}
                                >
                                    Add Athletes
                                    <AddIcon />
                                </Button>
                            </div>
                        </div>
                    </Paper>
                </div>
                <Button
                    variant="contained"
                    color="primary"
                    className={classes.saveButton}
                    onClick={handleSave}
                >
                    Save
                </Button>
            </Paper>
        </div>
    );
}
