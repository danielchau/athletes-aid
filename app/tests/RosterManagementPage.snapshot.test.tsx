import * as React from 'react'
import RosterManagementPage from '../src/components/RosterManagementPage'
import renderer from 'react-test-renderer'

let date: Date = new Date("2019-01-01");

const mockInjuryNote = {
    createdBy: "Test createdBy",
    createdOn: date,
    content: "Test content"
}

const mockEmergencyContact = {
    id: "Test id",
    name: "Test name",
    cellPhone: "Test cellPhone",
    homePhone: "Test homePhone",
    email: "Test email"
}

const mockInjury = {
    id: "Test id",
    active: true,
    createdOn: date,
    createdBy: "Test createdBy",
    teamName: "Test teamName",
    athleteName: "Test athleteName",
    injuryDate: date,
    isSportsRelated: true,
    eventType: "Test eventType",
    position: "Test position",
    sideOfBody: "Test sideOfBody",
    locationOnBody: "Test locationOnBody",
    injuryType: "Test injuryType",
    severity: 1,
    status: "Test status",
    mechanism: "Test mechanism",
    injuryDescription: "Test injuryDescription",
    otherNotes: [mockInjuryNote]
}

const mockAthleteProfile = {
    id: "Test id",
    profilePicture: "Test profilePicture",
    name: "Test name",
    birthdate: "Test birthdate",
    schoolYear: 1,
    gender: "Test gender",
    weight: 1,
    height: 1,
    email: "Test email",
    cellPhone: "Test cellPhone",
    homePhone: "Test homePhone",
    healthCardNumber: "Test healthCardNumber",
    emergencyContact: mockEmergencyContact,
    files: ["Test files"],
    injuries: [mockInjury]
}

const mockPagePermissions = {
    profiles: true,
    roster: true,
    logging: true,
    injuries: true,
    rosterManagement: true
}


const mockUserPermissions = {
    label: "Test label",
    pages: mockPagePermissions,
    canSeeSearchBar: true,
    canEditOtherProfiles: true,
    canSeeInjuryDetails: true
}

const mockUser = {
    athleteProfile: mockAthleteProfile,
    permissions: mockUserPermissions
}

const mockTeams = [{
    id: "Test id",
    name: "Test name",
    season: "Test season",
    athleteIds: ["Test athleteIds 1", "Test athleteIds 2"]
}]

it('RosterManagementPage renders correctly', () => {
  const tree = renderer.create(<RosterManagementPage state={false} teams={mockTeams} currentUser={mockUser} />).toJSON()
  expect(tree).toMatchSnapshot()
})