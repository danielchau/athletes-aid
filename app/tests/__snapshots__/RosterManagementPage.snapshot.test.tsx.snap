// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RosterManagementPage renders correctly 1`] = `
<div
  className="makeStyles-root-1 makeStyles-drawerClosed-3"
>
  <div
    className="MuiPaper-root MuiPaper-elevation1 makeStyles-introPaper-4 MuiPaper-rounded"
  >
    <p
      className="MuiTypography-root makeStyles-introText-6 MuiTypography-body1"
    >
      Edit one of your current teams:
    </p>
    <div
      className="MuiFormControl-root makeStyles-formControl-5"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
        data-shrink={false}
        id="team-select-label"
      >
        Team
      </label>
      <div
        className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
        onClick={[Function]}
      >
        <div
          aria-haspopup="listbox"
          aria-labelledby=" team-select"
          className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSelect MuiOutlinedInput-inputSelect"
          id="team-select"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          role="button"
          tabIndex={0}
        >
          <span
            dangerouslySetInnerHTML={
              Object {
                "__html": "&#8203;",
              }
            }
          />
        </div>
        <input
          onAnimationStart={[Function]}
          type="hidden"
          value=""
        />
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
        <fieldset
          aria-hidden={true}
          className="PrivateNotchedOutline-root-171 MuiOutlinedInput-notchedOutline"
          style={
            Object {
              "paddingLeft": 27,
            }
          }
        >
          <legend
            className="PrivateNotchedOutline-legend-172"
            style={
              Object {
                "width": 0.01,
              }
            }
          >
            <span
              dangerouslySetInnerHTML={
                Object {
                  "__html": "&#8203;",
                }
              }
            />
          </legend>
        </fieldset>
      </div>
    </div>
    <p
      className="MuiTypography-root makeStyles-introText-6 MuiTypography-body1"
    >
      or add a new one:
    </p>
    <button
      aria-describedby={null}
      aria-label="add"
      className="MuiButtonBase-root MuiIconButton-root makeStyles-introButton-7"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      title="Create new team"
      type="button"
    >
      <span
        className="MuiIconButton-label"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
          />
        </svg>
      </span>
      <span
        className="MuiTouchRipple-root"
      />
    </button>
  </div>
  <div
    className="MuiPaper-root MuiPaper-elevation1 makeStyles-paperContent-8 MuiPaper-rounded"
  >
    <div
      className="makeStyles-teamInfoContainer-9"
    >
      <div
        className="MuiFormControl-root MuiTextField-root makeStyles-textInput-10 MuiFormControl-marginNormal"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
          data-shrink={false}
          htmlFor="team-name"
          id="team-name-label"
        >
          Team Name
        </label>
        <div
          className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoFocus={false}
            className="MuiInputBase-input MuiOutlinedInput-input"
            disabled={false}
            id="team-name"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="(e.g. Women's Soccer)"
            required={false}
            type="text"
            value=""
          />
          <fieldset
            aria-hidden={true}
            className="PrivateNotchedOutline-root-171 MuiOutlinedInput-notchedOutline"
            style={
              Object {
                "paddingLeft": 8,
              }
            }
          >
            <legend
              className="PrivateNotchedOutline-legend-172"
              style={
                Object {
                  "width": 0.01,
                }
              }
            >
              <span
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "&#8203;",
                  }
                }
              />
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        className="MuiFormControl-root MuiTextField-root makeStyles-textInput-10 MuiFormControl-marginNormal"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
          data-shrink={false}
          htmlFor="season"
          id="season-label"
        >
          Season
        </label>
        <div
          className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoFocus={false}
            className="MuiInputBase-input MuiOutlinedInput-input"
            disabled={false}
            id="season"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="(e.g. 2019/2020)"
            required={false}
            type="text"
            value=""
          />
          <fieldset
            aria-hidden={true}
            className="PrivateNotchedOutline-root-171 MuiOutlinedInput-notchedOutline"
            style={
              Object {
                "paddingLeft": 8,
              }
            }
          >
            <legend
              className="PrivateNotchedOutline-legend-172"
              style={
                Object {
                  "width": 0.01,
                }
              }
            >
              <span
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "&#8203;",
                  }
                }
              />
            </legend>
          </fieldset>
        </div>
      </div>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-saveButton-21 MuiButton-containedPrimary"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          Create
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </button>
    </div>
    <hr
      className="MuiDivider-root makeStyles-contentDivider-11"
    />
    <div
      className="makeStyles-athleteContentContainer-12"
    >
      <div
        className="makeStyles-createPromptContainer-26"
      >
        <p
          className="MuiTypography-root makeStyles-createPrompt-27 MuiTypography-body1"
        >
          Please create a team or select an existing one before adding/modifying athletes.
        </p>
      </div>
    </div>
  </div>
</div>
`;
